### ISSUE TITLE
reddit.com: ads

### ISSUE THREAD

**Issue by HMChase** [uBlockOrigin/uAssets#13072]
### Prerequisites

- [X] I read and understood the [policy as to what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets)
- [X] I verified that the issue has not already been reported (use [this button](https://user-images.githubusercontent.com/585534/146582579-c32707a0-36a1-4cc5-ad50-83172c9f67a8.png) to find out)
- [X] I forced an update of my filter lists (**how to do this:** [click _"Purge all caches"_](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#purge-all-caches), then [click _"Update now"_](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#update-now))
- [X] I did not remove any of the [default filter lists](https://user-images.githubusercontent.com/585534/91296476-ab9d8c00-e76a-11ea-9fb9-87d6861bdf73.png), or I verified that the issue is not caused by removing any of the default lists
- [X] I did not enable external filter lists, or I verified that the issue still occurs without enabling external filter lists
- [X] I do not have custom filters/rules, or I verified that the issue still occurs without custom filters/rules
- [X] I am not using [another content blocker along uBO](https://twitter.com/gorhill/status/1033706103782170625) (also mind browser built-in blockers)
- [ ] I did not answer truthfully to **all** the above checkpoints

### URL address of the web page

`https://www.reddit.com/`

### Category

ads

### Describe the issue

tl;dr: I can hear video ads on reddit when scrolling, but cannot see them.

This is a new issue - I've only noticed it in the past few days on reddit. When I'm scrolling, I can hear sometimes ads playing in the background, but no visible ad is present. The ad stops playing when I scroll further, presumably when the blocked ad scrolls off the page. Reddit autoplays videos & gifs (but only one at a time), and if an ad is playing, it will prevent the real video from playing. Issue is occurring on Windows 10 PC. Windows and uBlock are up to date.

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

</details>

### Configuration

<details>

```yaml
uBlock Origin: 1.42.4
Chromium: 101
filterset (summary): 
  network: 82136
  cosmetic: 42855
  scriptlet: 16789
  html: 0
listset (total-discarded, last updated): 
  added: 
    https://static.couponcabin.com/prd/whitelist/couponcabin-cashback.edd0256e4272.txt: 40-1, 1d.15h.2m
    https://gitcdn.xyz/repo/NanoMeow/MDLMirror/master/hosts.txt: 0-0, 1d.9h.42m
    https://mirror.cedia.org.ec/malwaredomains/justdomains: 0-0, never
  default: 
    user-filters: 3-0, never
    easylist: 66297-51, 1d.15h.1m
    easyprivacy: 26926-121, 1d.15h
    plowe-0: 3676-805, 1d.15h.6m
    ublock-abuse: 75-0, 1d.15h.4m
    ublock-badware: 4140-92, 1d.14h.57m
    ublock-filters: 30955-87, 1d.14h.59m
    ublock-privacy: 212-2, 1d.15h.5m
    ublock-quick-fixes: 204-0, 1h.53m
    ublock-unbreak: 1783-40, 1d.14h.58m
    urlhaus-1: 8824-0, 1h.52m
filterset (user): [array of 3 redacted]
trustedset: 
  added: [array of 3 redacted]
  removed: 
    edge-scheme
switchRuleset: 
  added: [array of 2 redacted]
modifiedUserSettings: [none]
modifiedHiddenSettings: [none]
supportStats: 
  allReadyAfter: 38178 ms (selfie)
  maxAssetCacheWait: 775 ms
popupPanel: 
  blocked: 506
  blockedDetails: 
    reddit.com: 143
    alexametrics.com: 1
    facebook.net: 1
    gfycat.com: 4
    quantserve.com: 2
    redditmedia.com: 1
    redditstatic.com: 354
```
</details>

**Comment by stephenhawk8054** [comment]
Can you test this filter?

`reddit.com##[id^="t3"].promotedlink:upward(.rpBJOHq2PR60pnwJlUyP0 > div)`

**Comment by JohnyP36** [comment]
I could not reproduce this. 
###### By `this` I mean hearing ads, but not seeing them.

**Comment by stephenhawk8054** [comment]
> I could not reproduce this.
> ###### By `this` I mean hearing ads, but not seeing them.

This one is quite hard to reproduce. Basically:

1. You'll need new reddit and maybe a US IP (for easier to reproduce),
2. Turn on autoplay (in chromium, firefox and maybe reddit settings as well),
3. Turn on audio volume of 1 one the videos on reddit when you browse, because reddit mutes audio as default for all posts. From this step, every video after that will autoplay with audio.
4. Continue browsing, if you stop right at somewhere that the `promoted` post is (which is hidden by the filter `.promotedlink` in EasyList) and that post is a video ads, the audio of that post will automatically play.

What we are suspecting is that we need to hide the far parent node of that element (or `.rpBJOHq2PR60pnwJlUyP0 > div`) to not activate the promoted posts' videos, like in RraaLL's screenshot here:

`https://www.reddit.com/r/uBlockOrigin/comments/uftkwz/i_have_been_getting_audio_ads_while_on_reddit/i750xnk/?context=3`

Problem is we are not sure that `.rpBJOHq2PR60pnwJlUyP0` is the only name of the parent node or not. That's why we are waiting for users' feedbacks.

**Comment by stephenhawk8054** [comment]
> To react to you: I followed the steps, but I can not help you further in this as I can still not reproduce it (not yet).

Yeah I understand, it's not consistent and cannot always reproduce it.

**Comment by HMChase** [comment]
> Can you test this filter?
> 
> `reddit.com##[id^="t3"].promotedlink:upward(.rpBJOHq2PR60pnwJlUyP0 > div)`

I added this filter and it seems to have worked. I've been using reddit for about 45 minutes and haven't had this issue reoccur. I'll update the ticket if I see it happen again. Thank you!

**Comment by stephenhawk8054** [comment]
As the filter currently seems to work for others on reddit too, I think we can add it until further reports.
